### PR TITLE
[Unity][DistIR] LowerGlobalViewToLocalView

### PR DESCRIPTION
**Pull request by jinhongyii**
This PR adds a new pass called LowerGlobalViewToLocalView. 
It shards loop-level computation by rewriting TIR according to the sharding annotation on computational graph.

Introduction on what global view and local view are: https://thirsty-airplane-926.notion.site/DistIR-presentation-76d557b0c9344510b37bda790d5db3ec?pvs=4

### CODE REVIEW

**Comment by jinhongyii** [comment]
ready for review

File: src/relax/distributed/transform/lower_global_view_to_local_view.cc [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+      Expr new_func_body = this->VisitExpr(func_->body);
+      ObjectPtr<FunctionNode> new_func = make_object<FunctionNode>(*func_);
+      new_func->body = new_func_body;
+      builder_->UpdateFunction(gv, Function(new_func));
+    }
+    return builder_->GetContextIRModule();
+  }
+
+ private:
+  inline Array<DTensorStructInfo> ExtractDTensorStructInfo(Var var) {
```
Comment (yongwww) [inline src/relax/distributed/transform/lower_global_view_to_local_view.cc:376]: could it be a nested tuple (eg: `tup(dtensor, tup(dtensor, dtensor))`)? not sure if this case is disallowed

**Review by yongwww — COMMENTED** [review]
Overall it looks good to me! Thanks for sharing the do; it helped me a lot!

File: src/relax/distributed/transform/lower_global_view_to_local_view.cc (hunk 1 above)
Comment (jinhongyii) [reply src/relax/distributed/transform/lower_global_view_to_local_view.cc:376]: good catch. My current assumption is to disallow nested tuple. And we can surely support it if we find it's necessary to do so.

**Review by yongwww — APPROVED** [review]
LGTM

**Pull request merged**